src/main/scala/mesosphere/marathon/core/storage/StorageConfig.scala

```diff
@@ -29,10 +29,11 @@ import scala.collection.immutable.Seq
 import scala.concurrent.ExecutionContext
 import scala.concurrent.duration.{ Duration, _ }
 import scala.reflect.ClassTag
+// scalastyle:on
 
 sealed trait StorageConfig
```

Comment:
(1) mostly agree, but I had something different in mind:

StorageConfig itself not a "Scallop Option", its a typesafe "runtime edition" of the config, so we can read from `Config` later or `Scallop` now. Having this in the impl makes sense.

The storeCache() is defined only in `MarathonConf` sadly, otherwise, this only needs ZookeeperConf.

(2) I believe I put store under "storage"